projects/admin/src/app/record/detail-view/template-detail-view/template-detail-view.component.ts

```diff
@@ -0,0 +1,87 @@
+/*
+ * RERO ILS UI
+ * Copyright (C) 2020 RERO
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, version 3 of the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { RecordService } from '@rero/ng-core';
+import { Observable, of, Subscription } from 'rxjs';
+
+@Component({
+  selector: 'admin-template-detail-view',
+  templateUrl: './template-detail-view.component.html'
+})
+export class TemplateDetailViewComponent implements OnInit, OnDestroy {
+
+  /** Observable resolving record data */
+  record$: Observable<any>;
+
+  /** Observable of the imported record in marc format */
+  marc$: Observable<any>;
+
+  /** Record subscription */
+  private _recordObs: Subscription;
+
+  /** Resource type */
+  type: string;
+
+  /** Document record */
+  record: any;
+
+  /** Css class for dd in template */
+  ddCssClass = 'col-sm-9 mb-0';
+
+  /** Css class for dt in template */
+  dtCssClass = 'col-sm-3';
+
+  /** constructor
+   * @param _router - ActivatedRoute to get url parameters.
+   * @param _recordService - RecordService to the MARC version for the record.
+   */
+  constructor(
+    private _router: ActivatedRoute,
+    private _recordService: RecordService
+  ) { }
+
+  /** On init hook */
+  ngOnInit() {
+    this._recordObs = this.record$.subscribe((record: any) => {
+      this.record = record;
+      // only for imported record
+      if (record != null && record.metadata != null && this.record.metadata.pid == null) {
+        this.marc$ = this._recordService.getRecord(
+          this._router.snapshot.params.type, this.pid, 0, {
+          Accept: 'application/marc+json, application/json'
+        });
+      } else {
+        this.marc$ = of(null);
+      }
+    });
+  }
+
+    /** External identifier for imported record. */
+    get pid() {
+      if (this._router.snapshot && this._router.snapshot.params && this._router.snapshot.params.pid !== null) {
```

Comment:
Why testing the existence of `snapshot` and `snapshot.params` and not doing that for retrieving parameter `type`? I think the `snapshot` property is always set and `params`, too. But I'm not sure..

Reply:
We should ask @jma because I used the code in document-detail-view and he wrote it ^^

Reply:
Ok, but at line `65`, if snapshot or params is not existing, it will raise an error, so it's useless to check the existence in this method.